pkg/rke2/psp/psp.go

```diff
@@ -0,0 +1,443 @@
+package psp
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/rancher/spur/cli"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/policy/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/transport"
+)
+
+const (
+	defaultRetries          = 5
+	defaultWaitSeconds      = 5
+	defaultTimeout          = 30
+	k8sWrapTransportTimeout = 30
+)
+
+const (
+	kubeSystemNamespace = "kube-system"
+	cisAnnotationValue  = "resolved"
+
+	namespaceAnnotationBase               = "psp.rke2.io/"
+	namespaceAnnotationGlobalRestricted   = namespaceAnnotationBase + "global-restricted"
+	namespaceAnnotationGlobalUnrestricted = namespaceAnnotationBase + "global-unrestricted"
+	namespaceAnnotationSystemUnrestricted = namespaceAnnotationBase + "system-unrestricted"
+)
+
+// setGlobalUnrestricted sets the global unrestricted podSecurityPolicy with
+// the associated role and rolebinding.
+func setGlobalUnrestricted(ctx context.Context, cs *kubernetes.Clientset, ns *v1.Namespace) error {
+	if _, ok := ns.Annotations[namespaceAnnotationGlobalUnrestricted]; !ok {
+		if _, err := cs.RbacV1().ClusterRoles().Get(ctx, globalUnrestrictedPSPName, metav1.GetOptions{}); err != nil {
```

Comment:
Currently, this just swallows the error if it is anything other than IsNotFound. Those should probably get bubbled up instead. (This applies to multiple places)

Reply:
must-have. hopefully a small change and want to get error handling right